src/test/java/com/google/javascript/gents/singleTests/inner_types.js

```diff
@@ -0,0 +1,53 @@
+goog.module("foo.MyClass");
+
+class MyClass {
+  /**
+   * Constructor for MyClass
+   * @param {{type: !MyClass.InnerTypedefWithAssignment}} data
+   */
+  constructor(data) {
+    /** @export {!MyClass.InnerTypedefWithAssignment} type property*/
+    this.type = data.type;
+  }
+
+  /**
+   * equal function
+   * @param {!MyClass} otherData
+   * @return {boolean}
+   * @export
+   */
+  equals(otherData) {
+    return this.type.a === otherData.type.a;
+  }
+}
+
+/**
+ * @typedef {{
+ *     a: number,
+ *     b: number,
+ * }}
+ */
+MyClass.InnerTypedefWithAssignment = {};
+
+/**
+ * @typedef {{
+ *     a: number,
+ * }}
+ */
+MyClass.InnerTypedef;
+
+/**
+ * @typedef {{
+ *     a: {b: {c: number}},
+ * }}
+ */
+MyClass.InnerTypedefWithNestedTypes;
+
+/**
+ * @typedef {{
```

Comment:
We can consider down the road to change those to interfaces too, now that we have the capabilities.